[5.5.x] Fix error handling

bernardjkim opened the pull request:
More error handling fixes.

- Log pod not found error with log level WARN instead of DEBUG.
- Report last recorded packet loss percentage when probes fails.
- Previously reporting additional success probe even if checks failed.

Also fix `getNethealthAddr` by removing the field selector. Cannot rely on `spec.nodeName` to be equal to the advertised IP. Also `status.hostIP` seems to be unsupported as a field selector.

bernardjkim committed:
Fix getNethealthAddr

- Remove field selector. Cannot rely on spec.nodeName to be equalt to
advertised IP. Also status.hostIP seems to be unsupported as a field
selector.

bernardjkim committed:
Log not found error as WARNING instead of DEBUG

bernardjkim committed:
Report last recorded packet loss percentage when probe fails

a-palchikov commented on monitoring/nethealth.go:
```diff
@@ -260,6 +271,18 @@ func (c *nethealthChecker) isHealthy(peer string) (healthy bool, err error) {
+	return &pb.Probe{
+		Checker: name,
+		Detail: fmt.Sprintf("overlay packet loss for peer %s is higher than the allowed threshold of %.2f%%: %.2f%%",
+			peer, packetLossThreshold*100, packetLoss*100),
```
nit: you could percompute threshold:
```go
const thresholdPercent = packetLossThreshold*100
```

a-palchikov commented on monitoring/nethealth.go:
```diff
@@ -228,9 +231,17 @@ func (c *nethealthChecker) verifyNethealth(peers []string, reporter health.Repor
+		if err != nil {
+			return trace.Wrap(err)
 		}
+		packetLoss := data.packetLoss[len(data.packetLoss)-1]
```
Should take into account that `packetLoss` maybe empty.

r0mant commented on monitoring/nethealth.go:
```diff
@@ -142,18 +144,19 @@ func (c *nethealthChecker) check(ctx context.Context, reporter health.Reporter)
 	}
-	return fmt.Sprintf("http://%s:%d", pods.Items[0].Status.PodIP, c.NethealthPort), nil
+
+	return addr, trace.NotFound("unable to find local nethealth pod")
```
Nit: I would include the IP we were looking for (`c.AdvertiseIP`) in the message.

r0mant commented on monitoring/nethealth.go:
```diff
@@ -260,6 +271,18 @@ func (c *nethealthChecker) isHealthy(peer string) (healthy bool, err error) {
+func nethealthFailureProbe(name, peer string, packetLoss float64) *pb.Probe {
+	return &pb.Probe{
+		Checker: name,
+		Detail: fmt.Sprintf("overlay packet loss for peer %s is higher than the allowed threshold of %.2f%%: %.2f%%",
```
"peer" is too low-level as this is a user-facing message. Let's say "node".

r0mant reviewed (approved):
lgtm with a couple of comments + Dima's comments.

bernardjkim committed:
Address PR feedback

- Precompute threshold percent
- Check for empty pack loss
- Log advertised IP
- "peer" -> "node"
